framework/source/class/qx/ui/form/VirtualSelectBox.js

```diff
@@ -399,6 +433,17 @@ qx.Class.define("qx.ui.form.VirtualSelectBox",
       } else {
         return keyIdentifier;
       }
+    },
+
+
+    /**
+     * @param event {qx.event.type.Data} {@link qx.data.Array} change event.
+     * @private
+     */
+    __updateValue : function(event) {
+      var d = event.getData();
+      var old = (d.removed.length ? d.removed[0] : null);
+      this.fireDataEvent("changeValue", d.added[0], old);
```

Comment:
As it may come to the situation that some derived class wants to change the value type it would be good to also change this from a private to a protected member.